pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -456,25 +547,25 @@ func (meta *predicateMetadata) RemovePod(deletedPod *v1.Pod, node *v1.Node) erro
 	if deletedPodFullName == schedutil.GetPodFullName(meta.pod) {
 		return fmt.Errorf("deletedPod and meta.pod must not be the same")
 	}
-	meta.topologyPairsAntiAffinityPodsMap.removePod(deletedPod)
+	meta.podAffinityMetadata.topologyPairsAntiAffinityPodsMap.removePod(deletedPod)
 	// Delete pod from the matching affinity or anti-affinity topology pairs maps.
-	meta.topologyPairsPotentialAffinityPods.removePod(deletedPod)
-	meta.topologyPairsPotentialAntiAffinityPods.removePod(deletedPod)
+	meta.podAffinityMetadata.topologyPairsPotentialAffinityPods.removePod(deletedPod)
+	meta.podAffinityMetadata.topologyPairsPotentialAntiAffinityPods.removePod(deletedPod)
```

Comment:
Sorry I missed this, please create a removePod func for poAffinityMetadata. This will map to the Score.Extenders.RemovePod interface.